packages/babel-parser/src/plugins/flow.js

```diff
@@ -2061,6 +2061,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           return node.operator === "=";
 
         case "ParenthesizedExpression":
+        case "TypeCastExpression":
```

Comment:
Should also check for `TSTypeAssertion`.

Reply:
This is just for flow.
Arrow functions in conditional were broken in our TS parser since forever. I wanted to keep this PR small to fix the regression (fixing it in TS requires much more work)

Reply:
(hmm... GitHub should offer a facepalm reply)